### PR TITLE
chore: cull unused dependencies

### PR DESCRIPTION
**Pull request by joshuarli**
There seems to be quite a few unused dependencies, as well as stuff that wasn't cleaned up from going py2 -> py3.

Also, closes https://github.com/getsentry/snuba/pull/923 since `pylint` is removed.

A lot of this stuff is dev dependencies, currently not sure how production images are built but I'd bet that they have dev deps. It'd be nice to separate out - if there's any interest, I can follow up with that.

(It's quite a bit, see: https://github.com/getsentry/snuba/commit/88f821a1e6c54627eda9c16d13beca5bc61d0b54#diff-f4c265073ea4274a80eca85440875e56)

Another thought: it'd be nice to somehow get the docker build to use travis's pip cache. Currently, everything is being downloaded every time, and some wheels that need to be built are always rebuilt (uwsgi being the big one). Probably can save around a minute or two if this is figured out.

### CODE REVIEW

**Comment by fpacifici** [comment]
> A lot of this stuff is dev dependencies, currently not sure how production images are built but I'd bet that they have dev deps. It'd be nice to separate out - if there's any interest, I can follow up with that.

That would be awesome thanks. We have already discussed about that.
https://github.com/getsentry/snuba/pull/832#discussion_r389970087

File: .vscode/settings.json [hunk 1]
```diff
@@ -41,7 +41,6 @@
     "editor.formatOnSave": true
   },
 
-  "python.linting.pylintEnabled": false,
```
Comment (fpacifici) [inline .vscode/settings.json:44]: Does this default to false ?

Comment (joshuarli) [reply .vscode/settings.json:44]: I'm not sure, but there's no sense in having pylint at all since flake8 is being used and enforced in CI.

**Comment by joshuarli** [comment]
> > A lot of this stuff is dev dependencies, currently not sure how production images are built but I'd bet that they have dev deps. It'd be nice to separate out - if there's any interest, I can follow up with that.
> 
> That would be awesome thanks. We have already discussed about that.
> [#832 (comment)](https://github.com/getsentry/snuba/pull/832#discussion_r389970087)

Cool, I'll look into it when I have some spare cycles.

File: requirements.txt [hunk 2]
```diff
@@ -1,88 +1,52 @@
...
 sentry-relay>=0.5.7,<0.6.0
 sentry-sdk==0.13.5
-simplegeneric==0.8.1
 simplejson==3.15.0
-singledispatch==3.4.0.3
-traceback2==1.4.0
-traitlets==4.3.2
 typing-extensions==3.7.4.1
 unittest2==1.1.0
+traceback2==1.4.0
```
Comment (fpacifici) [inline requirements.txt:48]: I think you do not need to explicitly depend on this. It is already installed by unittest2 as a dependency

File: requirements.txt [hunk 3]
```diff
@@ -1,88 +1,52 @@
...
-Pygments==2.2.0
-pylint==1.9.2
-pycodestyle<2.6.0,>=2.5.0
 pyparsing==2.2.0
 pytest==5.2.4
 pytest-cov==2.5.1
-pytest-forked==1.1.3
 pytest-watch==4.2.0
-pytest-xdist==1.30.0
+colorama==0.3.9
```
Comment (fpacifici) [inline requirements.txt:35]: This list seems mostly alphabetically ordered, is there a reason to keep this out of order?

**Review by fpacifici — APPROVED** [review]
They seem ok, I checked all the removed ones. Please see my comments before merging.

File: requirements.txt (hunk 2 above)
Comment (joshuarli) [reply requirements.txt:48]: Yeah. `unittest2` as well:

```
$ pipdeptree -r -p traceback2
traceback2==1.4.0
  - unittest2==1.1.0 [requires: traceback2]
    - deprecation==2.0.3 [requires: unittest2]
```

My only hesitation would be that those were specifically == pinned in the past, but deprecation 2.0.3 resolves:

```
Successfully installed traceback2-1.4.0 unittest2-1.1.0
```

So I'll remove those two. There's probably a few leftovers like this in requirements.txt, not really worth hunting for them right now.

File: requirements.txt (hunk 3 above)
Comment (joshuarli) [reply requirements.txt:35]: Nope, but I can just reorder for aesthetics.

**Pull request merged**